### PR TITLE
Separable conv2d

### PR DESCRIPTION
**Pull request by anurag629**

### CODE REVIEW

**Comment by anurag629** [comment]
#8522

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py [hunk 1]
```diff
@@ -336,6 +336,35 @@ def test_dropout1d(
         assert u.shape == v.shape == w.shape
 
 
+@handle_test(
+    fn_tree="functional.experimental.separable_conv2d",
```
Comment (Aarsh2001) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py:340]: you are passing the wrong function tree here, and so the test fails at collection time. It should be `functional.ivy.experimental.separable_conv2d`.

**Review by Aarsh2001 — CHANGES_REQUESTED** [review]

**Comment by anurag629** [comment]
@Aarsh2001 How to resolve this error? I have not passed `kernel` variable then why this error?
Screenshot from 2023-01-17 21-34-58

**Comment by Aarsh2001** [comment]
Hey @anurag629, here are the following errors i see with the testing setup-:

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py [hunk 2]
```diff
@@ -336,6 +336,34 @@ def test_dropout1d(
         assert u.shape == v.shape == w.shape
 
 
+@handle_test(
+    fn_tree="functional.ivy.experimental.separable_conv2d",
+    ground_truth_backend="jax",
+    x_k_s_p=helpers.arrays_for_pooling(min_dims=2, max_dims=2, min_side=1, max_side=4),
```
Comment (Aarsh2001) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py:342]: Inside `arrays_for_pooling`, you are not generating the appropriate `array_dims` and so the `kernel` is not initialized at all. You are passing `max_dims = 2`. I suggest you look at the helper function inside the file, you'll get an idea.

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py [hunk 3]
```diff
@@ -336,6 +336,34 @@ def test_dropout1d(
...
+)
+def test_separable_conv2d(
+    *,
+    x_k_s_p,
+    test_flags,
+    backend_fw,
+    fn_name,
+    ground_truth_backend,
+):
+    dtype, x, stride, pad = x_k_s_p
```
Comment (Aarsh2001) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py:352]: You are not unpacking `kernel` here.

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py [hunk 4]
```diff
@@ -336,6 +336,34 @@ def test_dropout1d(
...
+    helpers.test_function(
+        ground_truth_backend=ground_truth_backend,
+        input_dtypes=dtype,
+        test_flags=test_flags,
+        fw=backend_fw,
+        fn_name=fn_name,
+        rtol_=1e-2,
+        atol_=1e-2,
+        x=x[0],
+        strides=stride,
```
Comment (Aarsh2001) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_layers.py:362]: You haven't passed `kernel` variable here as well.

File: ivy/functional/ivy/experimental/layers.py [hunk 5]
```diff
@@ -621,6 +621,77 @@ def dropout1d(
     )
 
 
+@to_native_arrays_and_back
+@handle_out_argument
+@handle_nestable
+def separable_conv2d(
+    x: Union[ivy.Array, ivy.NativeArray],
+    depthwise_filter: Union[int, Tuple[int], Tuple[int, int, int]],
+    pointwise_filter: Union[int, Tuple[int], Tuple[int, int, int]],
```
Comment (Aarsh2001) [inline ivy/functional/ivy/experimental/layers.py:630]: As you can see the function requires two filters, but at test time you are generating only one kernel. You should be generating two kernels/filters.

**Review by Aarsh2001 — CHANGES_REQUESTED** [review]

**Comment by Aarsh2001** [comment]
The tests are still failing, kindly have a look at the logs.

**Pull request closed without merging**